core/templates/dev/head/components/CorrectEditorDirective.js

```diff
@@ -0,0 +1,99 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the correct editor.
+ */
+
+// This directive controls an editor for selecting whether the answers
+// which fall under this answer group are correct or not. It also includes
+// 'Cancel' and 'Save' buttons which call respective 'onCancelCorrectEdit'
+// and 'onSaveCorrect' callbacks when called.
+oppia.directive('correctEditor', [
+  '$log', 'UrlInterpolationService', function($log, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        isEditable: '&isEditable',
+        onSaveCorrect: '&',
+        labelledAsCorrect: '='
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/' +
+        'correct_editor_directive.html'),
+      controller: [
+        '$scope', 'ResponsesService', 'stateInteractionIdService', function(
+            $scope, ResponsesService, stateInteractionIdService) {
+          $scope.savedCorrect = angular.copy($scope.labelledAsCorrect);
```

Comment:
Btw, these lines should go in init(), right?

Reply:
I followed the similar pattern as it is in RuleEditorDirective.js. Do we need init()? The labelledAsCorrect can never be null.
So I think may be  we should remove the init() function. What do you think ?

Reply:
Let's put all the "outside stuff" into $scope.init(), and then call $scope.init(). I think it's cleaner that way, otherwise you could end up with a lot of random stuff scattered throughout the directive controller. (We should tidy up the other directives similarly at some point.)

Reply:
Okay! . Done!